src/RADCodeGen.hs

```diff
@@ -0,0 +1,557 @@
+module RADCodeGen where
+  import Grammar
+  import LALR
+  import GenUtils (mapDollarDollar)
+  import RADTools (XGrammar(..), showItem, showProd, lhs, showRecognitionPoint, recognitionPoints, rhsAfterDot)
+  import RADStateGen
+  import Control.Monad
+  import Data.List
+  import Data.Maybe
+  import Data.Text (pack, unpack, replace)
+  import GHC.Arr ((!), indices)
+  
+
+  data ParserType = Normal | Monad | MonadLexer deriving (Eq, Show)
```

Comment:
You probably need more modes than that

Reply:
The `ParserType` enumeration only reflects the operating modes introduced via the presence of `%monad` and `%lexer` directives. These affect whether wrapper types are required (because of lexerWrapper / repeatTok etc.) and some other places of code generation.

Therefore, we would, at most, need four of these values. Currently, only `Normal` and `MonadLexer` is really supported by my implementation, and the other options (`Monad` and possible `LexerOnly`) still need to be implemented. I could add this on the TBD list if requested. But I haven't found any happy grammars which were using either of `%monad` or `%lexer` without also using the other. Therefore I decided to, at the moment, just support `Normal` and `MonadLexer`.